src/medal_widget/views.cljs

```diff
@@ -1,12 +1,39 @@
 (ns medal-widget.views
   (:require
    [re-frame.core :as re-frame]
+   [medal-widget.events :as events]
    [medal-widget.subs :as subs]
    ))
 
+(defn table-head [sort-key]
+  [:tr
+   [:th "S.no"]
+   [:th "Country"]
+   [:th [:span.gold-btn {:class (cond (= sort-key :gold) "medal-selected") :on-click #(re-frame/dispatch [::events/sort-key :gold])}]]
+   [:th [:span.silver-btn {:class (cond (= sort-key :silver) "medal-selected") :on-click #(re-frame/dispatch [::events/sort-key :silver])}]]
+   [:th [:span.bronze-btn {:class (cond (= sort-key :bronze) "medal-selected") :on-click #(re-frame/dispatch [::events/sort-key :bronze])}]]
+   [:th [:span.total-btn {:class (cond (= sort-key :total) "total-selected") :on-click #(re-frame/dispatch [::events/sort-key :total])} "Total"]]
+   ])
+
+(defn table-body [idx {:keys [code gold silver bronze total]}]
+  [:tr {:key idx} 
+   [:td (+ idx 1)]
+   [:td code]
+   [:td gold]
+   [:td silver]
+   [:td bronze]
+   [:td total]])
+ 
+
 (defn main-panel []
-  (let [name (re-frame/subscribe [::subs/name])]
+  (let [sort-key @(re-frame/subscribe [::subs/sort-key])
+        medals-data @(re-frame/subscribe [::subs/medals-data sort-key])]
```

Comment:
create a new subscription ::sorted-medals-data which makes it so the view doesn't need to deref and pass sort-key